src/rules/no-named-as-default.js

```diff
@@ -28,7 +35,15 @@ module.exports = {
         return;
       }
 
-      if (imports.has('default') && imports.has(defaultSpecifier[nameKey].name)) {
+      if (imports.has('default') && imports.has(analyzedName)) {
+
+        // #1594: the imported module exports the same thing via a default export and a named export
+        const namedExportInImportedModule = imports.reexports.get(analyzedName);
+        const defaultExportInImportedModule = imports.reexports.get('default');
+        if (defaultExportInImportedModule.getImport().path === namedExportInImportedModule.getImport().path
+          && defaultExportInImportedModule.local === namedExportInImportedModule.local) {
```

Comment:
```suggestion
        if (
          defaultExportInImportedModule.getImport().path === namedExportInImportedModule.getImport().path
          && defaultExportInImportedModule.local === namedExportInImportedModule.local
        ) {
```

Reply:
Apologies, but I don't believe that this particular line formatting is particularly better. The file is not Prettified anyway, and even if it was, then this line would be formatted even more differently than you suggest.

Reply:
Eventually, the comment is not relevant anymore, because of various changes in the final version.

Reply:
it's better because any time a `(` doesn't end a line, it should be on the same line as the corresponding `)`.

Reply:
This is the first time I hear such a rule, but I'm glad we don't have any outstanding formatting issues anymore :+1: